core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1020,9 +1020,21 @@ class TimeSeriesShard(val dataset: Dataset,
   }
 
   def shutdown(): Unit = {
+    // Don't release memory if the VM is actually shutting down. This can cause a segv if
+    // other threads are still accessing the memory. Everything gets freed when the process
+    // has completely exited anyhow. Note that this check isn't elegant or perfect.
+    val t = new Thread()
+    try {
+      Runtime.getRuntime().addShutdownHook(t);
```

Comment:
This method can be run in the context of a shutdown hook. Need to find out if another hook can be setup from a running shutdown hook and what the implications are.

Reply:
I didn't really want this change to be in this PR. Pushed to wrong branch...